RawCMS.Library/Lambdas/JsDispatcher.cs

```diff
@@ -28,23 +29,115 @@ public JsDispatcher(EntityService entityService)
             this.entityService = entityService;
         }
 
-        public override void Execute(string collection, ref JObject item, ref Dictionary<string, object> dataContext)
+        public void ExecuteInternal(string collection, ref JObject item, ref Dictionary<string, object> dataContext, PipelineStage stage, DataOperation Operation)
         {
+            var eventName = Stage.ToString() + Operation.ToString();
             var settings = this.entityService.GetByName(collection);
+
             if (settings != null)
             {
-                if (!string.IsNullOrEmpty(settings.PresaveScript))
+                var eventScript = settings.Events?[eventName];
+
+                if (eventScript != null)
                 {
-                    Dictionary<string, object> input = item.ToObject<Dictionary<string, object>>();
-
-                    Engine engine = new Engine((x) => { x.AllowClr(typeof(JavascriptRestClient).Assembly); x.AllowClr(typeof(JavascriptRestClientRequest).Assembly); });                    
-                    engine.SetValue("RAWCMSRestClient", Jint.Runtime.Interop.TypeReference.CreateTypeReference(engine, typeof(JavascriptRestClient)));
-                    engine.SetValue("RAWCMSRestClientRequest", Jint.Runtime.Interop.TypeReference.CreateTypeReference(engine, typeof(JavascriptRestClientRequest)));
-                    engine.SetValue("item", input);
-                    engine.Execute(settings.PresaveScript);
-                    item = JObject.FromObject(input);
+                    if (!string.IsNullOrEmpty(eventScript.ToString()))
+                    {
+                        Dictionary<string, object> input = item.ToObject<Dictionary<string, object>>();
+
+                        Engine engine = new Engine((x) => { x.AllowClr(typeof(JavascriptRestClient).Assembly); x.AllowClr(typeof(JavascriptRestClientRequest).Assembly); });
+                        engine.SetValue("RAWCMSRestClient", Jint.Runtime.Interop.TypeReference.CreateTypeReference(engine, typeof(JavascriptRestClient)));
+                        engine.SetValue("RAWCMSRestClientRequest", Jint.Runtime.Interop.TypeReference.CreateTypeReference(engine, typeof(JavascriptRestClientRequest)));
+                        engine.SetValue("item", input);
+                        engine.Execute(eventScript.ToString());
+                        item = JObject.FromObject(input);
+                    }
                 }
+                
             }
         }
     }
+
+    public class PreSaveEvent : JsDispatcher
```

Comment:
Create a single file for each class

Reply:
which folder you prefer to store this event.cs files? Lambdas?